deltacat/storage/main/impl.py

```diff
@@ -1,6 +1,15 @@
+"""
+Implementation of storage interface
+
+NOTE: THIS CURRENTLY DIVERGES FROM storage/interface.py!
+
+After this implementation is done, we will make storage/interface_v2.py. We then have to migrate
+  storage/iceberg/impl from storage/interface_v1.py to storage/interface_v2.py
```

Comment:
If you're only diverging by adding parameters, don't worry about revving the interface into an old and new version, since all of these APIs always had `*args` and `**kwargs` available, and the additional parameters should thus continue to flow through the old interface and just be unused by existing implementations.

Also, don't worry about keeping the iceberg implementation working - this will be discarded and turned into a metadata synchronization implementation.